lisa/operating_system.py

```diff
@@ -150,15 +196,35 @@ def type_name(cls) -> str:
     def name_pattern(cls) -> Pattern[str]:
         return re.compile(f"^{cls.type_name()}$")
 
-    def _install_packages(self, packages: Union[List[str]]) -> None:
+    def _install_packages(
+        self, packages: Union[List[str]], signed: bool = True
+    ) -> None:
+        raise NotImplementedError()
+
+    def _query_package(self, package: str, signed: bool = True) -> bool:
         raise NotImplementedError()
 
     def _initialize_package_installation(self) -> None:
         # sub os can override it, but it's optional
         pass
 
+    # _install_package_from_url
+    # Used if the package to be installed needs to be
+    # downloaded from a url first.
+    def _install_package_from_url(
```

Comment:
remove the starting underscore, it's a public method.

Reply:
fixed.